fmriprep/workflows/anatomical.py

```diff
@@ -420,52 +420,74 @@ def init_anat_template_wf(longitudinal, omp_nthreads, num_t1w, name='anat_templa
+    # 1. Template (only if several T1w images)
+    # 2. Reorient template to RAS, if needed (mri_robust_template may set to LIA)
     t1_reorient = pe.Node(Reorient(), name='t1_reorient')
 
-    lta_to_fsl = pe.MapNode(fs.utils.LTAConvert(out_fsl=True), iterfield=['in_lta'],
-                            name='lta_to_fsl')
-
     concat_affines = pe.MapNode(
-        ConcatAffines(3, invert=True), iterfield=['mat_AtoB', 'mat_BtoC'],
+        ConcatAffines(1 + min(num_t1w, 2), invert=True),
+        iterfield=['mat_AtoB', 'mat_BtoC'] if num_t1w > 1 else 'mat_AtoB',
         name='concat_affines', run_without_submitting=True)
 
+    if num_t1w > 1:
+        # 1a. Correct for bias field: the bias field is an additive factor
+        #     in log-transformed intensity units. Therefore, it is not a linear
+        #     combination of fields and N4 fails with merged images.
+        # 1b. Align and merge if several T1w images are provided
+        n4_correct = pe.MapNode(
+            N4BiasFieldCorrection(dimension=3, copy_header=True),
+            iterfield='input_image', name='n4_correct',
+            n_procs=1)  # n_procs=1 for reproducibility
+        t1_merge = pe.Node(
+            fs.RobustTemplate(auto_detect_sensitivity=True,
+                              initial_timepoint=1,      # For deterministic behavior
+                              intensity_scaling=True,   # 7-DOF (rigid + intensity)
+                              subsample_threshold=200,
+                              fixed_timepoint=not longitudinal,
+                              no_iteration=not longitudinal,
+                              transform_outputs=True,
+                              ),
+            mem_gb=2 * num_t1w - 1,
+            name='t1_merge')
+
+        lta_to_fsl = pe.MapNode(fs.utils.LTAConvert(out_fsl=True), iterfield=['in_lta'],
+                                name='lta_to_fsl')
+
+        def _set_threads(in_list, maximum):
+            return min(len(in_list), maximum)
+
+        workflow.connect([
+            (t1_conform, n4_correct, [('out_file', 'input_image')]),
+            (n4_correct, t1_merge, [('output_image', 'in_files')]),
+            (t1_conform, t1_merge, [
+                (('out_file', _set_threads, omp_nthreads), 'num_threads'),
+                (('out_file', add_suffix, '_template'), 'out_file')]),
+            (t1_merge, t1_reorient, [('out_file', 'in_file')]),
+            # Combine orientation and template transforms
+            (t1_merge, lta_to_fsl, [('transform_outputs', 'in_lta')]),
+            (lta_to_fsl, concat_affines, [('out_fsl', 'mat_BtoC')]),
+            (t1_reorient, concat_affines, [('transform', 'mat_CtoD')]),
+        ])
+    else:
+
+        def _get_first(in_list):
+            return in_list[0]
+
+        workflow.connect([
+            (t1_conform, t1_reorient, [(('out_file', _get_first), 'in_file')]),
+            (t1_reorient, concat_affines, [('transform', 'mat_BtoC')]),
```

Comment:
`t1_reorient` is not necessary if `t1_merge` is not used. Conformation orients to RAS, but for reasons I don't understand, merging sometimes changes the orientation. Then you can get rid of `concat_affines` and just use the conformation affine.